libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1807,26 +1802,15 @@ void TypeChecker::requireLValue(Expression const& _expression)
 
 void TypeChecker::typeError(SourceLocation const& _location, string const& _description)
```

Comment:
I don't think there's a strong need for the restriction, and we could have things like `typeError( )` moved into `ErrorReporter` in this change.